packages/sources/coinbase-prime/src/endpoint/balance.ts

```diff
@@ -0,0 +1,45 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { config } from '../config'
+import { httpTransport } from '../transport/balance'
+
+export const inputParameters = new InputParameters(
+  {
+    portfolio: {
+      required: true,
+      type: 'string',
+      description: 'The portfolio ID to query the balance of',
+    },
+    symbol: {
+      required: true,
+      type: 'string',
+      description: 'The symbol to return the balance for',
+    },
+    type: {
+      type: 'string',
+      description: 'The balance type to return',
+      default: 'total',
+      options: ['total', 'vault', 'trading'],
```

Comment:
maybe not relevant, but if `TOTAL` all caps was passed in, would this fail on invalid option?

Reply:
I believe options are case-sensitive so it would fail. Either GTM or EngOps usually test requests against the adapter before creating the job so a bad request should be caught ahead of time.